### PR TITLE
feat/SERV-12556-scenario-improvements

### PR DESCRIPTION
**Pull request by PageTMcEneaney**
## Context

At the moment, there is no option to add both data and error objects outside of the `body`. We'd like to add to support partial success graphql responses. See [example](https://www.apollographql.com/docs/react/data/error-handling/#graphql-error-policies). I want to make sure this is compatible with the parrot-friendly formatting of the scenarios so I think we should update the scenarios file to use or add a second example app

```js
{
  "data": {
    "getInt": 12,
    "getString": null
  },
  "errors": [
    {
      "message": "Failed to get string!"
      // ...additional fields...
    }
  ]
}
```

### Links

Issue: https://github.com/americanexpress/parrot/issues/210

#### Steps to Test 
Link local modules in `examples/friendly-pirate-app/package.json` to apply local changes
```
“parrot-friendly”: “file:../../packages/parrot-friendly”,
“parrot-graphql”: “file:../../packages/parrot-graphql”,
```

#### friendly-pirate-app
image

image

### CODE REVIEW

File: examples/friendly-pirate-app/scenarios.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  //   // graphql('/graphql', schema, {
+  //   //   Query: () => ({
+  //   //     Name: () => 'Jolly Roger',
+  //   //     Captain: () => 'Black Beard',
+  //   //   })
+  //   // })
+  //   graphql('/graphql', schema, () => ({
+  //     ShipLog: () => { data: 'response' }
+  //   }))
+  // });
```
Comment (PageTMcEneaney) [inline examples/friendly-pirate-app/scenarios.js:75]: I would appreciate some help with this mock, I want to prove that a graphql partial response is working as intended but was running into proxy issues.

File: packages/parrot-friendly/src/Mock.js [hunk 2]
```diff
@@ -32,6 +32,16 @@ export default class Mock {
...
+  data = value => {
+    this.structure.response.data = value;
+    return this;
+  };
+
+  errors = value => {
+    this.structure.response.errors = value;
+    return this;
+  };
+
```
Comment (PageTMcEneaney) [inline packages/parrot-friendly/src/Mock.js:44]: These changes should support a partial response for non-graphql mocks

File: packages/parrot-middleware/src/ParrotMiddleware.js [hunk 3]
```diff
@@ -36,6 +37,9 @@ class ParrotMiddleware extends Parrot {
     if (typeof body === 'object') {
       res.json(body);
     } else if (typeof body === 'undefined') {
+      if (resKeys.includes('data') || resKeys.includes('errors')) {
+        res.json(response);
+      }
```
Comment (PageTMcEneaney) [inline packages/parrot-middleware/src/ParrotMiddleware.js:42]: These changes should return a partial success response (outside of {body: {}} object) for graphql calls

File: examples/friendly-pirate-app/scenarios.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+// const schema = `
+// type Query {
+//   shiplog: [ShipLog]
+// }
+
+// type ShipLog {
+//   Name: String!
+//   Captain: String!
+// }
+// `;
```
Comment (Matthew-Mallimo) [inline examples/friendly-pirate-app/scenarios.js:29]: Can you remove the comments if they aren't needed / adding context

**Review by Matthew-Mallimo — CHANGES_REQUESTED** [review]

File: examples/friendly-pirate-app/scenarios.js (hunk 1 above)
Comment (Matthew-Mallimo) [reply examples/friendly-pirate-app/scenarios.js:75]: Can you explain the issues you were having?

**Pull request closed without merging**